cargo-dist/src/errors.rs

```diff
@@ -114,11 +133,29 @@ pub enum DistError {
 
     /// Use explicitly requested workspace builds, but had packages with custom feature settings
     #[error("precise-builds = false was set, but some packages have custom build features, making it impossible")]
-    #[help("these packages customized either features, no-default-features, or all-features: {packages:?}")]
+    #[diagnostic(help("these packages customized either features, no-default-features, or all-features: {packages:?}"))]
     PreciseImpossible {
         /// names of problem packages
         packages: Vec<String>,
     },
+    /// MSI with too many packages
+    #[error("The MSI {artifact_name} depends on multiple packages, which isn't yet supported")]
+    #[diagnostic(help("depends on {spec1} and {spec2}"))]
+    MultiPackageMSI {
+        /// Name of the MSI
+        artifact_name: String,
+        /// One of the pacakges
+        spec1: String,
+        /// A different package
+        spec2: String,
+    },
+    /// MSI with too few packages
+    #[error("The MSI {artifact_name} has no binaries associated with this")]
+    #[diagnostic(help("This should be impossible, you did nothing wrong, please file an issue!"))]
```

Comment:
Love "you did nothing wrong" as a user message. I want more software to tell me that.